### PR TITLE
feature/Santiago_Betancourth/solid

### PR DESCRIPTION
**Pull request by SantiagoBetancourth**

### CODE REVIEW

File: principles/solid/dependencyInjection.md [hunk 1]
```diff
@@ -1,22 +1,88 @@
...
+    class DatabaseService{
+        //...
+        void getData(){ //... }
+    }
+
+    class DataAccess {
+
+        private DatabaseService databaseService;
+
+        public DataAccess(DatabaseService databaseService){
```
Comment (zahydo) [inline principles/solid/dependencyInjection.md:24]: Here is using dependency Injection. Change the example

File: principles/solid/liskvsustitution.md [hunk 2]
```diff
@@ -1,22 +1,69 @@
...
+It declares that a subclass must be replaceable by its superclass, and if by doing this, the program crashes, we will be violating this principle.
+
+Adhering to this principle will confirm that our program has an easy-to-understand class hierarchy and reusable code.
 
 ### Bad example:
 
-[TODO]
+```JAVA:
+    public static void printNumSeats(Car[] arrayCars){
+    for (Car car : arrayCars) {
```
Comment (zahydo) [inline principles/solid/liskvsustitution.md]: Fix indentation

File: principles/solid/liskvsustitution.md [hunk 3]
```diff
@@ -1,22 +1,69 @@
...
-[TODO]
+For this method to comply with the principle, we will follow these principles:
 
-### Related principles
+- If the superclass (Car) has a method that accepts a parameter of the type of the superclass (Car), then its subclass
+  (Renault) should accept a type of the superclass (Car) or a type of the subclass (Renault) as its argument.
+- If the superclass returns a type of itself (Car), then its subclass (Renault) should either return a type of the
+  superclass (Car) or a type of the subclass (Renault).
+
+```JAVA:
```
Comment (zahydo) [inline principles/solid/liskvsustitution.md]: Where is this? http://univirtual.unicauca.edu.co/moodle/mod/page/view.php?id=77008

File: principles/solid/openclosed.md [hunk 4]
```diff
@@ -1,22 +1,93 @@
...
+    }
+
+    public static void printAveragePriceCar(Car[] arrayCars){
+        for (Car car : arrayCars) {
+            System.out.println(car.AverageCarPrice());
+        }
+    }
+```
+
+Each car extends the abstract class Car and implements the abstract method priceMedioCoche(). Thus, each car has its own implementation of the method AverageCarPrice(), so the method printAverageCarPrice() iterates the array of cars and only calls the method AverageCarPrice().
```
Comment (zahydo) [inline principles/solid/openclosed.md]: what is priceMedioCoche ? Did you forget to change what you copied?

**Review by zahydo — CHANGES_REQUESTED** [review]
Fix comments

File: principles/solid/dependencyInjection.md (hunk 1 above)
Comment (SantiagoBetancourth) [reply principles/solid/dependencyInjection.md:24]: Profe pero es el principio de Inyeccion de dependencias?

**Comment by zahydo** [comment]
Checked. Good Job

**Pull request closed without merging**